### ISSUE TITLE
More Verbose Output for Stochastic Runs?

### ISSUE THREAD

**Issue by herter4171** [idaholab/moose#16503]
## Reason

I'm helping Russell do 1E6 BISON runs on a boatload of cores on one of our clusters using the monte-carlo stuff in MOOSE.  I'm on pins and needles hoping the job doesn't go poorly, and the tail end of the logs is currently `Initializing equation system` followed by periodic printing of dots like what's below.

![image](https://user-images.githubusercontent.com/21989132/102280125-8dff5880-3efa-11eb-92c2-bf11cfa972f6.png)

## Design

I know this is a bit petty, but would it be possible to have something more verbose here that somehow indicates progress?

## Impact

Honestly, I don't have a clue as to what's going on under the hood, so I'm not sure.

**Comment by permcody** [comment]
I agree that an unknown number of dots that just come out is nearly as bad as having no output. There is no information here at all. We've been trying to go through and add more progress statements in places that take a long time for big simulations. Without specific input like this we can't improve so this is a start. We can always take a look at a problem like this and figure out what's taking so long so that we can improve it. Did this particular simulation eventually run?

**Comment by herter4171** [comment]
Sadly, no.  We boldly chose to go with a truckload of nodes, and it burned cores and printed dots all night.  Lesson learned there.  We've been able to do up to 1E6 cases on 5 nodes with 96 cores a pop.  I'm running on 6, 8, and 10 nodes of the same size with 1E5 cases to see where things start to break down.  In case it's not already in the docs, I had to set `ulimit -n 16384` to prevent crashes past around ten of our nodes.

**Comment by herter4171** [comment]
@permcody, I finished my scaling study.  As you can see on the plot, predicted run times held through 10 nodes still with 96 cores a pop.  With higher node/core counts, however, the run just continues on seemingly forever without raising any sort of error, so I've killed them after 10 minutes of excess run time for cost reasons.  Obviously, you guys have scaled up well past this node/core count before, so are there any gotchas we should be aware of regarding things like MPI settings or sysadmin stuff?  

I'd greatly appreciate some help here, because I'm not sure where else to turn with no errors happening.

![image](https://user-images.githubusercontent.com/21989132/102537968-acd82900-4079-11eb-8505-8de64f54e35f.png)

**Comment by permcody** [comment]
There are a number of different things that can affect scalability. The first and easiest is the amount of work you give each processor. In the MOOSE world, we recommend that you keep your nonlinear system local DOFS" around 20K or higher. You will see a printout of this information at the top of your run. It'll show you total DOFS and local DOFS (for processor zero). If you don't have adaptivity or anything else unique going on in your simulation, that number will stay constant. You can do some quick arithmetic to figure out your scaling limit by just taking the total DOFS and dividing by 20K.

The next big thing is your preconditioner. Some preconditions perform worse as you spread them out more. Some don't but in any case, there is always more communication which is one of the many factors on why scaling is never perfect. It is unusual to have it completely hang though. We have a lot of experience debugging hanging jobs. The general procedure is to use a script to get stack traces from all of the processes that are running. We have a script that can be adapted in the repository to aid in this task. We then look at the stack traces to see if there is any divergence. If there is, then there is a bug that needs to be resolved. While these are rare (they would wreak havoc on everything), they do find them occasionally.

Please share a little more detail on the size of your simulation and your preconditioner and we can rule those issues out.
